htdocs/compta/facture/class/facture-rec.class.php

```diff
@@ -998,57 +999,69 @@ function strikeIfMaxNbGenReached($ret)
 	 *
 	 *  WARNING: This method change temporarly context $conf->entity to be in correct context for each recurring invoice found.
 	 *
-	 *  @param	int		$restictoninvoiceid		0=All qualified template invoices found. > 0 = restrict action on invoice ID
+	 *  @param	int		$restrictioninvoiceid		0=All qualified template invoices found. > 0 = restrict action on invoice ID
 	 *  @param	int		$forcevalidation		1=Force validation of invoice whatever is template auto_validate flag.
 	 *  @return	int								0 if OK, < 0 if KO (this function is used also by cron so only 0 is OK)
 	 */
-	function createRecurringInvoices($restictoninvoiceid=0, $forcevalidation=0)
+	function createRecurringInvoices($restrictioninvoiceid=0, $forcevalidation=0)
 	{
-		global $conf, $langs, $db, $user;
+		global $conf, $langs, $db, $user, $hookmanager;
+
 
 		$error=0;
+		$nb_create=0;
 
 		// Load translation files required by the page
-        $langs->loadLangs(array("main","bills"));
+		$langs->loadLangs(array("main","bills"));
 
-		$nb_create=0;
+		$hookmanager->initHooks(array('createrecurringinvoices'));
 
 		$now = dol_now();
 		$tmparray=dol_getdate($now);
 		$today = dol_mktime(23,59,59,$tmparray['mon'],$tmparray['mday'],$tmparray['year']);   // Today is last second of current day
 
-		dol_syslog("createRecurringInvoices restictoninvoiceid=".$restictoninvoiceid." forcevalidation=".$forcevalidation);
+		dol_syslog("createRecurringInvoices restrictioninvoiceid=".$restrictioninvoiceid." forcevalidation=".$forcevalidation);
 
 		$sql = 'SELECT rowid FROM '.MAIN_DB_PREFIX.'facture_rec';
 		$sql.= ' WHERE frequency > 0';      // A recurring invoice is an invoice with a frequency
 		$sql.= " AND (date_when IS NULL OR date_when <= '".$db->idate($today)."')";
 		$sql.= ' AND (nb_gen_done < nb_gen_max OR nb_gen_max = 0)';
 		$sql.= ' AND suspended = 0';
 		$sql.= ' AND entity = '.$conf->entity;	// MUST STAY = $conf->entity here
-		if ($restictoninvoiceid > 0) $sql.=' AND rowid = '.$restictoninvoiceid;
+		if ($restrictioninvoiceid > 0)
+			$sql.=' AND rowid = '.$restrictioninvoiceid;
 		$sql.= $db->order('entity', 'ASC');
 		//print $sql;exit;
+		$parameters = array(
+			'entity' => $conf->entity,
```

Comment:
@bugness-chl 
entity is already in $conf, you can use "global $conf" in your hook